src/sonarplanet.ts

```diff
@@ -1,60 +1,56 @@
-const sonarplanetBackendUrl = '%%SONAR_BACK_URL%%'
-const register = '/register-to-notification'
+import * as SonarPlanetAPI from "./sonarplanetAPI"
+
 
 $(document).ready(() => {
+  SonarPlanetAPI.createAccountIfNeeded().then(
+    (response) => {
+      if ([201,200].indexOf(response.status) === -1) {
+        $('.alert.alert-danger').css('display', 'block')
+        $('.track-btn').prop('disabled', true)
+      }
+      return response
+    }, (err) => {
+      console.error('Error during account creation')
+      return err
+    }
+  )
   var form = document.getElementById("trackAddressForm") as HTMLFormElement;
   form.addEventListener('submit', event => {
-    trackAddressNow(event);
+    trackAddressNow(event)
   })
-});
+})
 
 let trackAddressNow = (event: Event) => {
-  event.preventDefault();
-  $('.alert').each((index: Number, alert: HTMLElement) => {
-    $(alert).css('display', 'none')
+  event.preventDefault()
+  resetAlerts()
+  var inputAddress = document.getElementById("trackAddress") as HTMLInputElement
+  SonarPlanetAPI.registerServiceWorker().then((serviceWorkerRegistration) => {
```

Comment:
Too many callbacks and nested code.
Let's try to reduce this code in the next sprint.